test/ibmq/test_filter_backends.py

```diff
@@ -17,19 +17,26 @@
 from qiskit.providers.ibmq import least_busy
 
 from ..ibmqtestcase import IBMQTestCase
-from ..decorators import requires_provider
+from ..decorators import requires_provider, requires_device
 
 
 class TestBackendFilters(IBMQTestCase):
     """Qiskit Backend Filtering Tests."""
 
-    @requires_provider
-    def test_filter_config_properties(self, provider):
+    @requires_device
+    def test_filter_config_properties(self, backend):
         """Test filtering by configuration properties"""
-        n_qubits = 20 if self.using_ibmq_credentials else 5
+        # Use the default backend as a reference for the filter.
+        provider = backend._provider
+        n_qubits = backend.configuration().n_qubits
 
         filtered_backends = provider.backends(n_qubits=n_qubits, local=False)
+
         self.assertTrue(filtered_backends)
+        for filtered_backend in filtered_backends:
+            with self.subTest(backend=filtered_backend):
+                self.assertEqual(n_qubits, backend.configuration().n_qubits)
+                self.assertFalse(backend.configuration().local)
```

Comment:
I believe it might be better to use `filtered_backend` for the asserts instead of `backend`. After running the test, it seems that `backend` actually refers to the backend passed by the decorator and not the one used by `subTest`. As a result, it seems that the test continually asserts the same backend as of now.

Reply:
Yes! Good catch, thanks :+1: